Assignment 2-MinruoWang.Rmd

```diff
@@ -0,0 +1,258 @@
+     vertex.label.color = "black", vertex.size = node.size)
+
+```
+
+
+
+## Part III  
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.  
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweenness centrality and dregree. **Who is the most central person in the network?**  
+
+Answer: Xudian Zhang is the most central person in the network in terms of betweenness centrality.  
+Lintong Li is the most central person in the network in terms of degree.
+
+### Data Cleaning
+```{r data cleaning}
+library(tidyr)
+library(dplyr)
+
+# load data
+hudk <- read.csv("hudk4050-classes.csv", header = TRUE, sep = ",")
+
+# format: turn names from factors to characters
+hudk$First.Name <- as.character(hudk$First.Name)
+hudk$Last.Name <- as.character(hudk$Last.Name)
+
+# format: write a function to capitalize the names
+CapStr <- function(x) {
+  s <- strsplit(x, " ")[[1]]
+  paste(toupper(substring(s, 1,1)), tolower(substring(s, 2)), sep="", collapse=" ")
+}
+# apply the function
+hudk$First.Name <- sapply(hudk$First.Name , CapStr)
+hudk$Last.Name <- sapply(hudk$Last.Name , CapStr)
+
+# combine first and last name
+hudk$person <- paste(hudk$First.Name, hudk$Last.Name)
+
+# drop the column of first and last name
+hudk <- subset(hudk, select = c(person, Class.1, Class.2, Class.3, Class.4, Class.5, Class.6))
+
+# format: revise the course names
+levels(hudk$Class.3)[match("G 5067",levels(hudk$Class.3))] <- "QMSSG5067"
+levels(hudk$Class.4)[match("G 5072",levels(hudk$Class.4))] <- "QMSSG5072"
+levels(hudk$Class.1)[match("4050",levels(hudk$Class.1))] <- "HUDK4050"
+levels(hudk$Class.2)[match("4125",levels(hudk$Class.2))] <- "HUDM4125"
+levels(hudk$Class.3)[match("5026",levels(hudk$Class.3))] <- "HUDM5026"
+levels(hudk$Class.4)[match("5126",levels(hudk$Class.4))] <- "HUDM5126"
+levels(hudk$Class.3)[match("QMSS-G5072",levels(hudk$Class.3))] <- "QMSSG5072"
+levels(hudk$Class.2)[match("IFSF4090002",levels(hudk$Class.2))] <- "ITSF4090"
+levels(hudk$Class.3)[match("EDPS4002001",levels(hudk$Class.3))] <- "EDPS4002"
+levels(hudk$Class.4)[match("EDPS4021001",levels(hudk$Class.4))] <- "EDPS4021"
+
+# convert factor into character
+hudk[] <- lapply(hudk, as.character)
+
+# remove the space
+hudk[2:7] <- as.data.frame(lapply(hudk[2:7],function(x){gsub(" ","",x)}))
+hudk
+```
+
+### Person-class matrix
+```{r matrix}
+# gather into long spreadsheet
+hudk_long <- gather(hudk, "class", "course", 2:7)
+hudk_long <- hudk_long[,-2]
+
+# delete the rows with no course
+hudk_long <- hudk_long[!(hudk_long$course==""), ]
+hudk_long$count <- 1
+
+# spread the class into column names
+hudk_wide <- spread(hudk_long, course, count)
```

Comment:
Nice job on data wrangling! You can omit HUDK 4050 class from the analysis since everyone in the dataset is taking this class.